impeller/entity/geometry/ellipse_geometry.cc

```diff
@@ -20,24 +33,52 @@ GeometryResult EllipseGeometry::GetPositionBuffer(
   auto& host_buffer = pass.GetTransientsBuffer();
   using VT = SolidFillVertexShader::PerVertexData;
 
-  Scalar radius = radius_;
+  Scalar half_width = stroke_width_ < 0
+                          ? 0.0
+                          : LineGeometry::ComputePixelHalfWidth(
+                                entity.GetTransform(), stroke_width_);
+  Scalar outer_radius = radius_ + half_width;
+  Scalar inner_radius = half_width <= 0 ? 0.0 : radius_ - half_width;
```

Comment:
I'm thinking that this might be easier to read as an if/else setting the values more directly in each code block than a series of ternary `?:` operations.